ksqldb-engine/src/main/java/io/confluent/ksql/security/KsqlPrincipal.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.security;
+
+import java.security.Principal;
+import java.util.Map;
+
+public interface KsqlPrincipal extends Principal {
+
+  Map<String, Object> getUserProperties();
```

Comment:
This like a good candidate for a default implementation returning an empty map.

Reply:
Done, though I left the no-op overrides in the two implementation classes (JaasPrincipal and DefaultKsqlPrincipal) for clarity. I can remove the redundancy if preferred.